osh/word_eval.py

```diff
@@ -232,6 +232,11 @@ def _ValueToPartValue(val, quoted, part_loc):
             s = val_ops.Stringify(val, loc.Missing)
             return part_value.String(s, quoted, not quoted)
 
+        elif case(value_e.List):
```

Comment:
On line 231 above, I guess we could add value_e.List` so Stringify is called, and then we can reuse the same error message?

It's a little subtle but we could add a comment to call that out

Reply:
Idk if the comment is OK as it is. 😅